www/js/app.js

```diff
@@ -311,7 +346,62 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
         apiStatusPanel.classList.add(apiPanelClass);
 
     }
-    
+
+    /**
+     * Queries Service Worker if possible to determine CACHE capability and returns an object with cache attributes
+     * If Service Worker is not available, the attributes of the memory cache are returned instead
+     * @returns {Promise} A Promise for an object with cache attributes 'type', 'description', and 'count'
```

Comment:
The `type` does not seem be currently used. I suppose it's for some future usage?
If you have something in mind, it's fine : I just want to check it's not a mistake

Reply:
Ah, my editor (vscode) shows me details for the Promise interface as if it were a valid type. But I think it's heavily orientated towards TypeScript, where strong typing is mandatory. I'll just change this to `{Object}` and make clear that it's a promise for an object in the description. We do have some custom non-JS types in use -- e.g. we have `@type Integer` elsewhere, whereas strictly there is no such primitive JS type, and we should use `{Number}`.

Reply:
I was not mentioning this annotation (which seems right), but the attribute`type` of the returned object (next to `description` and `count`)

Reply:
Ah, OK. I'm not using "type" as a technical term. It's the name of a key, so if I assign this returned object to a variable like `cache` I can use `cache.type`, `cache.description` and `cache.count`. The string `cache.type` could be reported as the strings `cacheAPI`, or `memory`, or `custom` or even `none`, depending on context and user options. Are you OK with `type` as the name of this object key? Type and description are subtly different. Type would be `cacheAPI` and description is `Cache API` with a space and initial capital (a future DEV might want a fuller description, for example).

Reply:
Yes, you're right I don't now use it. I did before, and that code has been cut in the simplification. Should I keep?

Reply:
I'm OK with the `type` name, and understand the purpose. But I did not see where this attribute was used.
It's not a big issue to keep it if you think it might be useful in the future

Reply:
OK, I'll keep it. It should be useful for extensibility and I'm averse to having data structures that use descriptions as primary keys (from working with relational databases)!